itest/rust/src/object_tests/call_deferred_test.rs

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+use crate::framework::itest;
+use godot::obj::WithBaseField;
+use godot::prelude::*;
+use godot::task::{SignalFuture, TaskHandle};
+use std::ops::DerefMut;
+
+const ACCEPTED_NAME: &str = "touched";
+
+#[derive(GodotClass)]
+#[class(init,base=Node2D)]
+struct DeferredTestNode {
+    base: Base<Node2D>,
+}
+
+#[godot_api]
+impl DeferredTestNode {
+    #[signal]
+    fn test_completed(name: StringName);
+
+    #[func]
+    fn accept(&mut self) {
+        self.base_mut().set_name(ACCEPTED_NAME);
+    }
+
+    fn to_assertion_task(&self) -> TaskHandle {
+        assert_ne!(
+            self.base().get_name().to_string(),
+            ACCEPTED_NAME,
+            "accept evaluated synchronously"
+        );
+
+        let test_will_succeed: SignalFuture<(StringName,)> =
+            Signal::from_object_signal(&self.to_gd(), "test_completed").to_future();
```

Comment:
Can you use `self.signals().test_completed()` here?

Or a particular reason to use untyped signals?

Reply:
Good question. I think initially I wanted less internal api dependencies in case something is changed. But I already changed this in #5 and don't see a benefit anymore. I'll replace this.

This also showed that with signals, `&mut self` is required.